app/src/main/java/com/sun/noteapp/ui/todonote/ToDoNoteActivity.kt

```diff
@@ -0,0 +1,237 @@
+package com.sun.noteapp.ui.todonote
+
+import android.app.DatePickerDialog
+import android.app.TimePickerDialog
+import android.content.Context
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.MenuItem
+import android.view.View
+import androidx.appcompat.app.AlertDialog
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.ItemTouchHelper
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.sun.noteapp.R
+import com.sun.noteapp.data.model.Note
+import com.sun.noteapp.data.source.local.NoteDatabase
+import com.sun.noteapp.ui.base.BaseDialog
+import com.sun.noteapp.ui.textnote.ImageColorDialog
+import com.sun.noteapp.ui.textnote.LabelAdapter
+import com.sun.noteapp.utils.*
+import kotlinx.android.synthetic.main.activity_to_do_note.*
+import kotlinx.android.synthetic.main.toolbar_text_note.*
+import java.text.SimpleDateFormat
+import java.util.*
+
+class ToDoNoteActivity : AppCompatActivity(),
+    View.OnClickListener {
+
+    private val labelAdapter = LabelAdapter()
+    private val date = Calendar.getInstance()
+    private var note: Note? = null
+    private var listLabel = listOf<String>()
+    private var remindTime = NONE
+    private var id = 0
+    private var status = NoteDatabase.UNHIDE
+    private var color = 0
+    private var password = NONE
+    private val toDos = mutableListOf<Pair<String, Boolean>>()
+    private var itemTouchHelper: ItemTouchHelper? = null
+    private val simpleDateFormat =
+        SimpleDateFormat(DATETIME_FORMAT, Locale.getDefault())
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_to_do_note)
+        initView()
+        initData()
+    }
+
+    private fun initView() {
+        setSupportActionBar(toolbarTitle)
+        supportActionBar?.setDisplayShowTitleEnabled(false)
+        toolbarBottom.inflateMenu(R.menu.bottom_option_menu)
+        toolbarBottom.setOnMenuItemClickListener { item: MenuItem? ->
+            openDialog(item?.itemId)
+        }
+        imageButtonHeaderColorTextNote.setOnClickListener(this)
```

Comment:
initViews() đang khá nhiêu code em cũng có thể tách các setOnClickListener(this) ra 1 phương thức
Mỗi phương thức nên chỉ tầm chục dòng code đổ lại thôi

Reply:
vâng